File: Source/CookItNow.Business/Models/Ingredient.cs
using System;
using System.Collections.Generic;

namespace CookItNow.Business.Models
{
    public class Ingredient 
    {
        public long Id { get; set; }
		
        public int SubrecipeId { get; set; }
		
        public string Name { get; set; }
		
        public Quantity Quantity { get; set; }
		
        public Ingredient Replacement { get; set; }
		
        public IList<string> Requirements { get; set; }
    }
}

File: Source/E133.Web/app/shared/models/ingredient.ts
import {Quantity, QuantityDto} from "shared/models/quantity";

export class Ingredient implements IngredientDto {
    id: number;
    subrecipeId: number;
    name: string;
    quantity: Quantity;
    requirements: string;
    replacements: Ingredient[];
    state: string;

    constructor();
    constructor(dto: IngredientDto);
    constructor(dto?: IngredientDto) {
        Object.assign(this, dto);

        if (dto) {
            this.quantity = new Quantity(dto.quantity);
            this.replacements = (dto.replacements || []).map(replacementDto => new Ingredient(replacementDto));
        }

        this.state = "";
    }
}
export interface IngredientDto {
    id: number;
    subrecipeId: number;
    name: string;
    quantity: QuantityDto;
    requirements: string;
    replacements: IngredientDto[];
}
